src/connection.rs

```diff
@@ -569,7 +590,14 @@ pub enum Error {
     /// The total message payload data size exceeds the configured maximum.
     MessageTooLarge { current: usize, maximum: usize },
     /// The connection is closed.
-    Closed
+    Closed(Option<CloseReason>),
+}
+
+/// Reason for closing the connection.
+#[derive(Debug)]
+pub struct CloseReason {
+    code: u16,
+    descr: Option<String>,
```

Comment:
I suppose you want to access these fields

```suggestion
    pub code: u16,
    pub descr: Option<String>,
```